### PR TITLE
fix(markdown): decode HTML entities in Markdown exports

### PR DESCRIPTION
**Pull request by davideg**
Thank you for this project! It's super helpful, but I noticed that Markdown exports contain HTML escape code entities like `&amp;` and  `&gt;`. I've never developed in Swift, but I decided it should be an easy enough fix. Luckily the existing functionality already existed for RTF export and just needed to be extracted into a utility function.

I used GitHub Copilot to create the changes but reviewed them carefully. I realize editing the `project.pbxproj` is more dangerous (and I really don't know what I'm doing there), but it was necessary to add an additional test file. I'm happy to remove those changes if you think the tests should live elsewhere.

----------------

## Summary

This PR fixes Markdown export to decode HTML entities (e.g. `&amp;` --> `&`).

This change extracts the existing HTML entity decoder from the RTF converter into a shared String.htmlDecoded property in ExportSupport.swift.

It applies the shared decoder to Markdown output after remaining HTML tags have been stripped. This ordering ensures encoded angle brackets such as &lt;example&gt; remain visible instead of being mistaken for HTML tags and removed.

It reuses the shared decoder in the RTF converter and remove its private duplicate implementation.

It adds regression coverage for named, decimal, and hexadecimal entities, including ampersands, encoded angle brackets, copyright symbols, and supplementary Unicode characters.

It registers the new Markdown export tests with the Xcode test target. **This required modifying `project.pbxproj`**

~Edit: I should mention the local tests passed and `make build` and `make run` were successful, but I wasn't able to figure out how to grant the Debug app Full Disk Access in order to make `make test-formats` work.~

Edit2: `make test-formats` passes ✨ 

## Related issues

N/A

## Checklist

- [x] Commits follow [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Every commit is signed off (`git commit -s`) per [DCO](../CONTRIBUTING.md#developer-certificate-of-origin-dco). By signing off, I certify the [Developer Certificate of Origin](https://developercertificate.org/) and agree that my contribution will be released under the project's [GPLv3 license](../LICENSE).
- [x] `make build` succeeds locally
- [ ] If UI changed: tested the happy path and edge cases in the running app
- [x] If new format/export logic: `make test-formats` passes
- [x] Added myself to `CONTRIBUTORS.txt` (if not already there)
- [x] No em-dashes in code comments or commit messages
- [ ] Did not change `project.pbxproj`, `*.xcconfig`, `*.entitlements`, or `.github/**` unless strictly necessary (these require owner review)

### CODE REVIEW

File: Apple Notes Exporter/Apple Notes Exporter/ExportSupport.swift [hunk 1]
```diff
@@ -41,6 +41,54 @@ extension String {
             .replacingOccurrences(of: "'", with: "&#39;")
     }
 
+    /// Decode common HTML entities to their character equivalents
+    var htmlDecoded: String {
```
Comment (kzaremski) [inline Apple Notes Exporter/Apple Notes Exporter/ExportSupport.swift:45]: I like the use of a string extension here. There are other opportunities for functions with one input and one output to become extensions that I will look into later.

**Review by kzaremski — APPROVED** [review]
